R/layer-sf.R

```diff
@@ -35,6 +35,18 @@ LayerSf <- ggproto("LayerSf", Layer,
         self$mapping$geometry <- as.name(geometry_col)
       }
     }
+
+    # automatically determine the legend type
+    if (is.na(self$show.legend) || self$show.legend == TRUE) {
+      if (is_sf(data)) {
+        if (sf_geometry_type(data) %in% c("POINT", "MULTIPOINT"))
+          self$geom_params$legend <- "point"
+        else if (sf_geometry_type(data) %in% c("LINESTRING", "MULTILINESTRING",
+                                               "CIRCULARSTRING", "COMPOUNDCURVE",
+                                               "CURVE", "MULTICURVE"))
+          self$geom_params$legend <- "line"
```

Comment:
Probably you can use `sf_types` instead of a raw list of sf types.

https://github.com/tidyverse/ggplot2/blob/64248088462cc3114d0eb4e30beafd2550d19284/R/geom-sf.R#L300-L306